### PR TITLE
Change how we handle defaults.

### PR DESCRIPTION
**Pull request by jnb**
* Only allow a None to be returned for an optional field if that field is x-nullable.
* Instantiate a missing field to its default value if provided.

### CODE REVIEW

File: bravado_core/unmarshal.py [hunk 1]
```diff
@@ -126,18 +126,12 @@ def unmarshal_object(swagger_spec, object_spec, object_value):
             type(object_value), object_value))
 
     object_spec = deref(object_spec)
-    required_fields = object_spec.get('required', [])
 
     result = {}
     for k, v in iteritems(object_value):
         prop_spec = get_spec_for_prop(
             swagger_spec, object_spec, object_value, k)
-        if v is None and k not in required_fields and prop_spec:
```
Comment (jnb) [inline bravado_core/unmarshal.py]: I'd be OK to revert this hunk if you think it's not worth the amount of pain it might cause?

Comment (sjaensch) [inline bravado_core/unmarshal.py]: I think the change makes sense conceptually, not sure how many people depend on being allowed to send `None` for optional fields though.

File: tests/unmarshal/unmarshal_object_test.py [hunk 2]
```diff
@@ -459,8 +470,10 @@ def test_non_required_none_value(empty_swagger_spec, property_type):
                                          nullable=False,
                                          property_type=property_type)
     value = {'x': None}
-    result = unmarshal_object(empty_swagger_spec, content_spec, value)
-    assert result == {'x': None}
+    # This should fail because the field is not nullable
+    with pytest.raises(SwaggerMappingError) as excinfo:
+        unmarshal_object(empty_swagger_spec, content_spec, value)
+    assert 'is a required value' in str(excinfo.value)
```
Comment (sjaensch) [inline tests/unmarshal/unmarshal_object_test.py]: This is a misleading error message, as the field is not required. We just don't allow `None` as value anymore.

**Comment by jnb** [comment]
OK, I scoped this down a bit.  Now the behavior is just to add a default field value in case of a missing field.  If there's isn't a default field specified, then we add a None (as before).

**Comment by sjaensch** [comment]
I've released version 4.7.3 with this change.

**Pull request merged**